### DISCUSSION TITLE
How to use a query result as a variable ?

### DISCUSSION THREAD

**Discussion by RitikaDesai98** [hasura/graphql-engine#9397] [Question] [answered]
Hello

I am using Hasura with my Flutter Application..

I have 2 tables: `tasks` and `categories`

`tasks` comprises of `id`, `task_name`, `category_id`, `status`.
`category` comprises of `id`, `category_name`, `user_id`, `color`.

What I want to do is get the name of the category that the task belongs to using the category_id

What I thought of is:

```
query getTasks($user_id: String!) {
  tasks(where: user_id: {_eq: $user_id}}, order_by: {created_at: desc}) {
    category_id
    name
  }
  category_by_pk(id: tasks['category_id']){
    name
  }
}
```

The part that is `tasks['category_id']` being passed as a query variablele is giving an error

Any idea how can I do this?

Thanks in advance

**Comment by adas98012** [comment, marked as answer]
@RitikaDesai98 
You should be able to create relationship between the "tasks" and "category" tables (with category_id as a foreign key) and then fetch "category" fields in a single query (via the relationship). 

Here is the reference [document](https://hasura.io/docs/latest/schema/postgres/table-relationships/create/).
